content/chainguard/chainguard-enforce/how-to-install-chainctl.md

```diff
@@ -84,17 +84,231 @@ You should receive output similar to the following.
+GitVersion:    0.1.42
+GitCommit:     badf10d2d89c83c6f1366008ffc3936d1e4c36f4
 GitTreeState:  clean
-BuildDate:     2022-12-05T14:04:10Z
-GoVersion:     go1.19.3
+BuildDate:     2022-12-12T22:10:55Z
+GoVersion:     go1.19.4
 Compiler:      gc
 Platform:      darwin/arm64
 ```
 
 If you received different output, check your bash profile to make sure that your system is using the expected PATH. 
 
+### Verifying the `chainctl` binary with Cosign and Rekor
+
+All `chainctl` binaries are released with [keyless signatures using Cosign](https://edu.chainguard.dev/open-source/sigstore/cosign/an-introduction-to-cosign/#keyless-signing). You can verify the signature for a `chainctl` binary using the `cosign` tool directly, or by calculating the SHA256 hash of the release and finding the corresponding Rekor transparency log entry.
+
+The following steps will work for both `curl` and `brew` installations of `chainctl`.
+
+#### Verifying `chainctl` using Cosign keyless signatures
+
+To verify a `chainctl` binary using Cosign, first ensure you have the latest versin of Cosign installed by following our [How to Install Cosign guide](https://edu.chainguard.dev/open-source/sigstore/cosign/how-to-install-cosign/).
+
+Next, run `chainctl version --json` to output the version information. You should receive output like the following:
+
+```json
+{
+  "gitVersion": "0.1.42",
+  "gitCommit": "badf10d2d89c83c6f1366008ffc3936d1e4c36f4",
+  "gitTreeState": "clean",
+  "buildDate": "2022-12-12T22:10:55Z",
+  "goVersion": "go1.19.4",
+  "compiler": "gc",
+  "platform": "darwin/arm64"
+}
+```
+
+Now set an environment variable to the version shown in the `gitVersion` field, which is `0.1.42` in this example.
+
+```sh
+CHAINCTL_VERSION="0.1.42"
+```
+
+Run the following commands to create two URL environment variables. These URLs will point to the correct version of the `chainctl` signature and public certificate files respectively:
+
+```sh
+SIGNATURE_URL="https://dl.enforce.dev/chainctl/${CHAINCTL_VERSION?}/chainctl_$(uname -s | tr '[:upper:]' '[:lower:]')_$(uname -m).sig"
+CERTIFICATE_URL="https://dl.enforce.dev/chainctl/${CHAINCTL_VERSION?}/chainctl_$(uname -s | tr '[:upper:]' '[:lower:]')_$(uname -m).cert.pem"
+```
+
+Check your environment variables for correctness using `echo`:
+
+```sh
+echo "$SIGNATURE_URL\n$CERTIFICATE_URL\n$CHAINCTL_VERSION"
+```
+
+You should receive output like the following, which shows the correct version, operating system, and CPU architecture for your system values in the computed URLs:
+
+```
+https://dl.enforce.dev/chainctl/0.1.42/chainctl_darwin_arm64.sig
+https://dl.enforce.dev/chainctl/0.1.42/chainctl_darwin_arm64.cert.pem
+0.1.42
+```
+
+Now you can verify the binary by running the following `cosign verify-blob` command:
+
+```sh
+COSIGN_EXPERIMENTAL=1 cosign verify-blob --signature $SIGNATURE_URL --certificate $CERTIFICATE_URL $(which chainctl)
+```
+
+You should receive the following output:
+
+```
```

Comment:
This is the 0.1.44 one in case you are interested

```
tlog entry verified with uuid: 801f8e482e178f739569d4bb96c70f25096ed4646fc2e1d264ed81bd3b97e1e5 index: 9106000
Verified OK
```